### ISSUE TITLE
Android 12 support for Firebase messaging

### ISSUE THREAD

**Issue by stianjensen** [firebase/firebase-android-sdk#3423]
The latest version of firebase-messaging does not work with Android 12. A fix has been implemented, but it's now been at least 3 months without a new release of firebase-messaging, which means it's still not possible to use FCM (a Google product) while targeting the latest version of Android (also a google product).

The original issue was closed and locked, while waiting for the release to be pushed:
https://github.com/firebase/firebase-android-sdk/issues/3115

If this is not the right place to discuss the release plan for a new version of firebase-messaging, can anyone direct me to where that might be?

**Comment by tareq3** [comment]
yes. needs the fix ASAP

**Comment by argzdev** [comment]
Thanks for bringing this up, @stianjensen. I'll notify an engineer and see when this can be released. Internally tracked at b/218830909.

**Comment by Eleftheria-99** [comment]
I am having the same issue here, always receiving crashes about this in Firebase Crashlytics + my app users cannot get their notifications

**Comment by argzdev** [comment]
Hi @tareq3 & @Eleftheria-99. Just to get more details, may I ask what version of firebase-messaging are you using? And also at what android version are your users experiencing the crashes? Thanks!

**Comment by Eleftheria-99** [comment]
Hi @argzdev the details that you asked : 

Firebase version :   
       `  // Firebase SDK
        // Import the BoM for the Firebase platform
        implementation platform('com.google.firebase:firebase-bom:26.5.0')

        implementation 'com.google.firebase:firebase-analytics-ktx'
        implementation 'com.google.firebase:firebase-auth-ktx'
        implementation 'com.google.firebase:firebase-config-ktx'
        implementation 'com.google.firebase:firebase-dynamic-links-ktx'
        implementation 'com.google.firebase:firebase-perf-ktx' // Firebase Performance monitoring
        implementation 'com.google.firebase:firebase-messaging-ktx'
        implementation 'com.google.firebase:firebase-crashlytics-ndk'
        implementation 'com.google.firebase:firebase-analytics-ktx'
     `
All the crashes are happening in Android 12! The users basically, they never see the push notifications, when the app is closed or in the background ! 
In every other Android version the push notifications are working perfectly fine !

**Comment by argzdev** [comment]
Thanks for the details, @Eleftheria-99. Our engineers have responded that this issue has already been fixed since with version 21.1.0. 

Currently, it looks like with version BoM 26.5.0, you're using version 21.0.1 for firebase-messaging. Could you try 21.1.0 or higher to see if fixes the issue?

Relevant code:
```
private static int getPendingIntentFlags(int var0) {
        return VERSION.SDK_INT >= 23 ? 1140850688 : 1073741824;
    }
```

In the code snippet of firebase-messaging. The value of 1140850688 is the bitwise OR of the two flags. FLAG_ONE_SHOT | FLAG_IMMUTABLE = 1140850688, which is the value in the FCM SDK. This should be fixed with versions 21.1.0 or higher.

Could you confirm if you're using the said version @stianjensen, @tareq3?

**Comment by Eleftheria-99** [comment]
Thank you a lot for the help ! Now it is fixed and it is working fine !

**Comment by EmanAhmed** [comment]
I'm using this version of android firebase

**Comment by EmanAhmed** [comment]
I'm using this version of android firebase and can't receive any notifications on android 12.
```
  // Import the BoM for the Firebase platform
  implementation platform('com.google.firebase:firebase-bom:29.1.0')
  // Declare the dependency for the Firebase Authentication library
  // When using the BoM, you don't specify versions in Firebase library dependencies
  implementation 'com.google.firebase:firebase-auth'
  // firebase core
  implementation 'com.google.firebase:firebase-core'
  // Firebase analytics
  implementation 'com.google.firebase:firebase-analytics'
  //Firebase cloud messaging.
  implementation 'com.google.firebase:firebase-messaging'


```

**Comment by argzdev** [comment]
Hi @EmanAhmed, your issue seems to be a different one, please file a new ticket for your issue with the necessary details so we can assists you further.

Also given that the main issue here is has already been fixed since version 21.1.0, I'll be closing this issue now.
